src/dishka/integrations/grpcio.py

```diff
@@ -30,37 +48,45 @@ def inject(func: Callable[P, RT]) -> Callable[P, RT]:
     )
 
 
-class DishkaInterceptor(grpc.ServerInterceptor):  # type: ignore[misc]
+class GrpcioProvider(Provider):
+    provides = (
+        from_context(Message, scope=Scope.REQUEST),
+        from_context(ServicerContext, scope=Scope.REQUEST),
+        from_context(ServicerContext, scope=Scope.SESSION),
```

Comment:
Sounds like a problem. We cannot have same factory in both scopes. But, if you do transition APP->REQUEST implicitly skipping SESSION, then context is available in both SESSION and REQUEST